Yesod/Auth/OAuth2.hs

```diff
@@ -87,15 +98,15 @@ authOAuth2Widget widget name oauth getCreds = AuthPlugin name dispatch login
         tm <- getRouteToParent
         render <- lift getUrlRender
         return oauth
-            { oauthCallback = Just $ encodeUtf8 $ render $ tm url
+            { oauthCallback = Just $ unsafeFromText $ render $ tm url
```

Comment:
Does it make sense to wrap this up in a `YesodOAuth2Exception`?

Reply:
Personally, I thought not because it is unexpected enough that the disruption of an `error` would be preferable. In practice, I don't see this ever blowing up because my understating is `render . tm` would either produce a valid URI or blow up itself first.